dashboard/app/controllers/script_levels_controller.rb

```diff
@@ -328,6 +312,57 @@ def present_level
     render 'levels/show', formats: [:html]
   end
 
+  # Returns a set of sections for the current user, and a filter set of which of
+  # those sections use the given script_id. For teachers, looks at sections they
+  # teacher, for students it's sections they are in.
+  def user_sections(script_id)
+    if current_user.try(:teacher?)
+      sections = current_user.sections.select{|s| s.deleted_at.nil?}
+    elsif current_user.try(:student?)
+      sections = current_user.sections_as_student.select{|s| s.deleted_at.nil?}
+    end
+
+    if sections.nil? || sections.empty?
+      return [[], []]
+    end
+
+    script_sections = sections.select{|s| s.script.try(:name) == script_id}
+    [sections, script_sections]
+  end
+
+  def stage_hidden?(current_user, script_level)
```

Comment:
Is `stage_hidden?` meant to depend on `current_user`? Note that it seems not to. Assuming not, any reason this parameter can't be removed?

Also for `get_hidden_stage_ids` method below.